### PR TITLE
Feature/all podcast feeds

### PR DESCRIPTION
**Pull request by jdwitten**
* This tries to get all podcasts for all 3 streams and maps them to a new model for displaying with an image.
* Sorts podcasts by date

### CODE REVIEW

File: iOS Client/HTTPClient/MBClient.swift
Comment (AlexRamey) [inline iOS Client/HTTPClient/MBClient.swift:8]: LOL at these pictures man . . .

File: iOS Client/HTTPClient/MBClient.swift
Comment (AlexRamey) [inline iOS Client/HTTPClient/MBClient.swift]: can we reuse the fields defined above?

Comment (AlexRamey) [reply iOS Client/HTTPClient/MBClient.swift]: I notice there's an inconsistency with http vs https for mockingCast. It should be https, same goes for pz

File: iOS Client/Info.plist
Comment (AlexRamey) [inline iOS Client/Info.plist]: I don't think this is necessary. Fireside supports `https` so we can just add that to the urls defined in our client above

File: iOS Client/Info.plist
Comment (AlexRamey) [inline iOS Client/Info.plist]: This file seems kind of janked up with regard to the spacing. Since mbird and youtube were already here, an we don't need fireside, you can just revert this file

File: iOS Client/HTTPClient/MBClient.swift
Comment (AlexRamey) [reply iOS Client/HTTPClient/MBClient.swift]: https works for the fireside endpoints, we should add https to our url constants for fireside ones

File: iOS Client/HTTPClient/MBClient.swift
Comment (AlexRamey) [inline iOS Client/HTTPClient/MBClient.swift]: nice

File: iOS Client/HTTPClient/MBClient.swift
Comment (AlexRamey) [inline iOS Client/HTTPClient/MBClient.swift]: very cool

File: iOS Client/Models/MBPodcast.swift
Comment (AlexRamey) [inline iOS Client/Models/MBPodcast.swift:32]: hmmmm, we need a DRYer way to do this . . . Maybe just make the image's type `Data` instead of UIImage (since `Data` is codable) and keep this one struct

File: iOS Client/PodcastsController/PodcastDetailViewController.swift
Comment (AlexRamey) [inline iOS Client/PodcastsController/PodcastDetailViewController.swift]: this branch will need to be rebased onto master once you merge the other PR

Comment (AlexRamey) [reply iOS Client/PodcastsController/PodcastDetailViewController.swift]: (or possibly merged . . .)

**Review by AlexRamey — CHANGES_REQUESTED** [review]

File: iOS Client/HTTPClient/MBClient.swift
Comment (AlexRamey) [inline iOS Client/HTTPClient/MBClient.swift:55]: nice, much cleaner

File: iOS Client/Models/MBPodcast.swift
Comment (AlexRamey) [reply iOS Client/Models/MBPodcast.swift:32]: dude, these are like identical models, anyway we can get away with just having one? 

Looks like pubDate is a date on DisplayablePodcast and that's the only difference, could just be a lazy property on the first struct that gets derived from the string date

File: iOS Client/PodcastsController/MBPodcastsViewController.swift
Comment (AlexRamey) [inline iOS Client/PodcastsController/MBPodcastsViewController.swift:49]: this is fine, but in general I usually live on the edge in the table view methods. I'd rather throw in an `!` than handle some error case that will never happen

File: iOS Client/PodcastsController/PodcastDetailViewController.swift
Comment (AlexRamey) [inline iOS Client/PodcastsController/PodcastDetailViewController.swift]: nice, very performant 👍

File: iOS Client/PodcastsController/PodcastDetailViewController.swift
Comment (AlexRamey) [inline iOS Client/PodcastsController/PodcastDetailViewController.swift]: these numbers will need to be formatted in mm:ss format, maybe the NumberFormatter can help . . .?

File: iOS Client/PodcastsController/PodcastDetailViewController.swift
Comment (AlexRamey) [reply iOS Client/PodcastsController/PodcastDetailViewController.swift]: hmmmmmm, we should make sure `totalDuration != 0`

File: iOS Client/PodcastsController/PodcastDetailViewController.swift
Comment (AlexRamey) [inline iOS Client/PodcastsController/PodcastDetailViewController.swift]: I don't get why this protocol exists? I thought we had addressed this earlier . . .

File: iOS Client/PodcastsController/PodcastsCoordinator.swift
Comment (AlexRamey) [inline iOS Client/PodcastsController/PodcastsCoordinator.swift]: this should be explicitly marked as the `PodcastHandler` method, which should really be called something like `PodcastViewControllerDelegate` if you decide to stick with delegation for the callback. I would say it's a no brainer, but with UDF in the equation I could see a desire to fire an action and let that come back down. Up to you. Delegation seems to be much more direct

File: iOS Client/PodcastsController/PodcastsCoordinator.swift
Comment (AlexRamey) [inline iOS Client/PodcastsController/PodcastsCoordinator.swift]: this shouldn't exist

Comment (AlexRamey) [reply iOS Client/PodcastsController/PodcastsCoordinator.swift]: as discussed previously

File: iOS Client/PodcastsController/PodcastsCoordinator.swift
Comment (AlexRamey) [inline iOS Client/PodcastsController/PodcastsCoordinator.swift]: `PodcastHandler` and `PodcastPlayerDelegate` protocol definitions should be swapped in terms of placement.

Think about how table view works. Right next to table view, they define `TableViewDataSource` and `TableViewDelegate` protocols. And the table view has properties of those types. Then outsiders can conform to those protocols and interface with table view.

In this case, the coordinator has a property of type `PodcastPlayerDelegate`. That protocol should be defined in this file. Then, the Podcast View Controller could implement it. Similarly, the Podcast View controller file should hold the definition of `PodcastHandler`. 

However, again, all of this is a moot point b/c we shouldn't have two-way delegation going on. This coordinator should be calling functions directly on the view controller. We don't need protocols for that direction. However, if you want to setup a callback scheme from the view controller back up to the coordinator, then delegation is the move for that direction. You'd define some delegate protocol in the view controller file and then conform to it in the coordinator file and directly set the view controller's delegate to self (the coordinator).

Think of delegation as a solution to when you need a callback. 

Remember, we know all about the table view, so we can tell it to do whatever we want without delegation. But the table view has no idea who we are, so we set its delegate property in order to receive callbacks. All the table view knows is that we conform to its delegate protocol.

With these thoughts in mind, I think this code can be made much simpler, which I thought was done earlier on b/c I'm pretty sure we had this discussion, but maybe GitHub is being weird.

File: iOS Client/Store/MBPodcastsStore.swift
Comment (AlexRamey) [inline iOS Client/Store/MBPodcastsStore.swift]: `indx` is a bit strange. I've seen `idx` before . . . not a big deal

File: iOS Client/Store/MBPodcastsStore.swift
Comment (AlexRamey) [inline iOS Client/Store/MBPodcastsStore.swift]: Hmm, I go back and forth. There may be some value in splitting our DTO (data transfer object) model from our domain model . . .

File: iOS Client/Models/MBPodcast.swift
Comment (AlexRamey) [reply iOS Client/Models/MBPodcast.swift:32]: Hmm, I'm actually not sure. Maybe if we rename MBPodcast to be called `PodcastDTO` (data transfer object) and rename DisplayablePodcast to be `Podcast`, I'll be okay with this

**Review by AlexRamey — CHANGES_REQUESTED** [review]

File: iOS Client/PodcastsController/PodcastsCoordinator.swift
Comment (jdwitten) [reply iOS Client/PodcastsController/PodcastsCoordinator.swift]: Yeah man so I just never merged those changes back into this branch, but it should be good to go now if you want to give it a rereview. Not sure which of these comments are still relevant so I would say give it a good look over the whole thing

File: iOS Client/Store/MBPodcastsStore.swift
Comment (jdwitten) [reply iOS Client/Store/MBPodcastsStore.swift]: My original intent for splitting it was for the images, but now that we are just storing the name of the image I think it might be ok to consolidate to one model

File: iOS Client/PodcastsController/MBPodcastsViewController.swift
Comment (AlexRamey) [inline iOS Client/PodcastsController/MBPodcastsViewController.swift:37]: nice

File: iOS Client/PodcastsController/MBPodcastsViewController.swift
Comment (AlexRamey) [inline iOS Client/PodcastsController/MBPodcastsViewController.swift:62]: nice

File: iOS Client/PodcastsController/PodcastTableViewCell.swift
Comment (AlexRamey) [inline iOS Client/PodcastsController/PodcastTableViewCell.swift:13]: whitespace, doesn't really matter

File: iOS Client/Store/MBPodcastsStore.swift
Comment (AlexRamey) [inline iOS Client/Store/MBPodcastsStore.swift:20]: this is some intense s*** man, super BA looking.

File: iOS Client/Store/MBPodcastsStore.swift
Comment (AlexRamey) [inline iOS Client/Store/MBPodcastsStore.swift]: very classy use of `when`

File: iOS Client/Store/XMLParsingDelegate.swift
Comment (AlexRamey) [inline iOS Client/Store/XMLParsingDelegate.swift:24]: you can one line this with `if elementName == "enclosure", let url = attributeDict["url"] { . . . }` I think . . . Certainly can do it if you're willing to put the let clause first

Comment (AlexRamey) [reply iOS Client/Store/XMLParsingDelegate.swift:24]: Question: What's the point of this? Wouldn't this just get overwritten below by the `guid` case of `didEndElement`?

File: iOS Client/Models/MBPodcast.swift
Comment (AlexRamey) [inline iOS Client/Models/MBPodcast.swift]: I like having two different models.

I think we should call this one `PodcastDTO` and `DisplayablePodcast` can be called `Podcast`.

I started this project by prefixing everything with `MB` b/c you had to use a custom prefix (usually 2 or 3 letters) in Objective-C world to avoid namespace collisions, but in swift that is no longer the case . . . 🤦‍♂️

**Review by AlexRamey — CHANGES_REQUESTED** [review]
Looks great.

My biggest critiques are
1) Rename MBPodcast to PodcastDTO and DisplayablePodcast to Podcast
2) Make sure the XML parse logic is sound

File: iOS Client/Store/XMLParsingDelegate.swift
Comment (jdwitten) [reply iOS Client/Store/XMLParsingDelegate.swift:24]: Yeah so this is a little artifact that the rss feeds are not structured the same way. 2 of the feeds have the url as fields on the xml element, and one has its own element called `guid` with the url. So the feeds that have `enclosure` don't have `guid` so they don't get overwritten. I think we might need to do some reexamination of this logic but I think it's good to get in for now

Comment (AlexRamey) [reply iOS Client/Store/XMLParsingDelegate.swift:24]: ah okay, seems reasonable. And I see you only have access to this `attributeDict` on the `didStartElement` delegate method.

**Review by AlexRamey — APPROVED** [review]
nice, I feel good about having a DTO model and a domain model. This is what we're trending toward on lidl

**Pull request merged**
